.stylelintrc-colors.js

```diff
@@ -0,0 +1,24 @@
+            /hsla{0,1}\(/i,
+            /hwb\(/i,
+            /gray\(/i,
+            // uses of web color names are not allowed (with 'white' and 'black' being the only exceptions)
+            /[^\$-\w](lavender|thistle|plum|violet|orchid|fuchsia|magenta|mediumorchid|mediumpurple|blueviolet|darkviolet|darkorchid|darkmagenta|purple|indigo|darkslateblue|slateblue|mediumslateblue|pink|lightpink|hotpink|deeppink|palevioletred|mediumvioletred|lightsalmon|salmon|darksalmon|lightcoral|indianred|crimson|firebrick|darkred|red|orangered|tomato|coral|darkorange|orange|yellow|lightyellow|lemonchiffon|lightgoldenrodyellow|papayawhip|moccasin|peachpuff|palegoldenrod|khaki|darkkhaki|gold|cornsilk|blanchedalmond|bisque|navajowhite|wheat|burlywood|tan|rosybrown|sandybrown|goldenrod|darkgoldenrod|per|chocolate|saddlebrown|sienna|brown|maroon|darkolivegreen|olive|olivedrab|yellowgreen|limegreen|lime|lawngreen|chartreuse|greenyellow|springgreen|mediumspringgreen|lightgreen|palegreen|darkseagreen|mediumseagreen|seagreen|forestgreen|green|darkgreen|mediumaquamarine|aqua|cyan|lightcyan|paleturquoise|aquamarine|turquoise|mediumturquoise|darkturquoise|lightseagreen|cadetblue|darkcyan|teal|lightsteelblue|powderblue|lightblue|skyblue|lightskyblue|deepskyblue|dodgerblue|cornflowerblue|steelblue|royalblue|blue|mediumblue|darkblue|navy|midnightblue|snow|honeydew|mintcream|azure|aliceblue|ghostwhite|whitesmoke|seashell|beige|oldlace|floralwhite|ivory|antiquewhite|linen|lavenderblush|mistyrose|gainsboro|lightgray|silver|darkgray|gray|dimgray|lightslategray|slategray|darkslategray)/i,
+            /^(lavender|thistle|plum|violet|orchid|fuchsia|magenta|mediumorchid|mediumpurple|blueviolet|darkviolet|darkorchid|darkmagenta|purple|indigo|darkslateblue|slateblue|mediumslateblue|pink|lightpink|hotpink|deeppink|palevioletred|mediumvioletred|lightsalmon|salmon|darksalmon|lightcoral|indianred|crimson|firebrick|darkred|red|orangered|tomato|coral|darkorange|orange|yellow|lightyellow|lemonchiffon|lightgoldenrodyellow|papayawhip|moccasin|peachpuff|palegoldenrod|khaki|darkkhaki|gold|cornsilk|blanchedalmond|bisque|navajowhite|wheat|burlywood|tan|rosybrown|sandybrown|goldenrod|darkgoldenrod|per|chocolate|saddlebrown|sienna|brown|maroon|darkolivegreen|olive|olivedrab|yellowgreen|limegreen|lime|lawngreen|chartreuse|greenyellow|springgreen|mediumspringgreen|lightgreen|palegreen|darkseagreen|mediumseagreen|seagreen|forestgreen|green|darkgreen|mediumaquamarine|aqua|cyan|lightcyan|paleturquoise|aquamarine|turquoise|mediumturquoise|darkturquoise|lightseagreen|cadetblue|darkcyan|teal|lightsteelblue|powderblue|lightblue|skyblue|lightskyblue|deepskyblue|dodgerblue|cornflowerblue|steelblue|royalblue|blue|mediumblue|darkblue|navy|midnightblue|snow|honeydew|mintcream|azure|aliceblue|ghostwhite|whitesmoke|seashell|beige|oldlace|floralwhite|ivory|antiquewhite|linen|lavenderblush|mistyrose|gainsboro|lightgray|silver|darkgray|gray|dimgray|lightslategray|slategray|darkslategray)/i
```

Comment:
this is to capture the instances when a rule's value start with `<colorname>`.

e.g.,

```css
color: colorname;
// We don't have to worry about the space after `:`.
// stylelint trims out whitespaces around a rule's `value` before running the test
```

Reply:
Rather than specifying this ourselves, could we just tap into https://github.com/stylelint/stylelint/tree/master/lib/rules/color-named instead and say `never`?

Reply:
not if we want to have exceptions - in this case `black` and `white`

Reply:
Yea that was my first attempt. But if we want to have `black` and `white` as exceptions then we can't use Stylelint's `color-named ` 😭

Reply:
Do we, though? Using $black and $white seems pretty okay in everything except the variables file?